.github/workflows/upload_all.yml

```diff
@@ -1,26 +1,132 @@
-name: Upload to all stores
-on: [workflow_dispatch]
+name: Upload
+
+on:
+  workflow_dispatch:
+    inputs:
+      targets:
+        type: choice
+        description: Where to upload
+        options:
+        - all
+        - google-play
+        - app-store
+
 jobs:
-  call_upload_google_play:
-    name: Upload to Google Play
-    uses: ./.github/workflows/upload_google_play.yml
-    secrets:
-      POLYPOD_POLYPEDIA_REPORT_URL: ${{ secrets.POLYPOD_POLYPEDIA_REPORT_URL }}
-      POLYPOD_POLYPEDIA_REPORT_AUTHORIZATION: ${{ secrets.POLYPOD_POLYPEDIA_REPORT_AUTHORIZATION }}
-      POLYPOD_ANDROID_UPLOAD_KEYSTORE_FILE: ${{ secrets.POLYPOD_ANDROID_UPLOAD_KEYSTORE_FILE }}
-      POLYPOD_ANDROID_UPLOAD_KEYSTORE_PASS: ${{ secrets.POLYPOD_ANDROID_UPLOAD_KEYSTORE_PASS }}
-      POLYPOD_ANDROID_UPLOAD_KEY_ALIAS: ${{ secrets.POLYPOD_ANDROID_UPLOAD_KEY_ALIAS }}
-      POLYPOD_ANDROID_STORE_SERVICE_ACCOUNT_JSON: ${{ secrets.POLYPOD_ANDROID_STORE_SERVICE_ACCOUNT_JSON }}
+  build_test_upload:
+    name: Build, test, and upload
+    runs-on: macos-11
+    steps:
+      - name: Fail if uploading from non-release branch
+        if: |
+          github.ref != 'refs/heads/main'
+          && startsWith(github.ref, 'refs/heads/release') != true
+          && startsWith(github.ref, 'refs/heads/prod4pod-1426') != true
+        run: exit 1
+      - run: echo "Uploading ${{ github.ref }}"
+      - name: Set upload targets
+        run: |
+          if [[ ${{ github.event.inputs.targets }} =~ all|google-play ]]; then
+              echo "Uploading to Google Play"
+              echo "POLYPOD_UPLOAD_GOOGLE_PLAY=true" >> $GITHUB_ENV
```

Comment:
OK, I see what you're doing here, turning a regex match (which GHA does not have) into a Boolean environment variable (which you can use either in step conditional expressions or in shell).